raiden/tests/integration/long_running/test_integration_events.py

```diff
@@ -525,10 +526,17 @@ def run_test_secret_revealed_on_chain(
         token_network_address, app0, deposit - amount, [], app1, deposit + amount, []
     )
 
-    with gevent.Timeout(10):
-        wait_for_state_change(
-            app2.raiden, ContractReceiveSecretReveal, {"secrethash": secrethash}, retry_interval
-        )
+    try:
+        with gevent.Timeout(10):
```

Comment:
Here it is better to wait on the block number, the timeout and lock expiration are not synchronized at all, also this should raise an exception describing what went wrong:

https://github.com/raiden-network/raiden/blob/1b8f7f71c2b4a9c376f02eed54a334d037e50644/raiden/utils/timeout.py#L19-L44

Reply:
Follow-up issue to do this in several places: #4524